app/lib/schema-data.ts

```diff
@@ -73,26 +96,59 @@ export type ExampleFiles = {
   name: string
   content: object
 }
+export type GitContentDataResponse = {
+  name: string
+  path: string
+  type: string
+  content?: string
+  children?: GitContentDataResponse[]
+}
 
 export async function loadSchemaExamples(
-  path: string
+  path: string,
+  ref: string
 ): Promise<ExampleFiles[]> {
-  const dirPath = path.substring(0, path.lastIndexOf('/') + 1)
+  const dirPath = path.substring(0, path.lastIndexOf('/'))
   const schemaName = path.substring(path.lastIndexOf('/') + 1)
-  const exampleFiles = (
-    await readdir(join(dirname(require.resolve('@nasa-gcn/schema')), dirPath))
-  ).filter(
+  const exampleFiles = (await getContentTree(dirPath, ref)).filter(
     (x) =>
-      x.startsWith(`${schemaName.split('.')[0]}.`) &&
-      x.endsWith('.example.json')
+      x.name.startsWith(`${schemaName.split('.')[0]}.`) &&
+      x.name.endsWith('.example.json')
   )
+
   let result: ExampleFiles[] = []
-  exampleFiles.forEach(async (exampleFile) => {
-    const example = await loadJson(join(dirPath, exampleFile))
+  for (const exampleFile of exampleFiles) {
+    const exPath = join(dirPath, '/', exampleFile.name)
+    const example = await loadContentFromGithub(exPath)
     result.push({
-      name: exampleFile.replace('.example.json', ''),
+      name: exampleFile.name.replace('.example.json', ''),
       content: example,
     })
-  })
+  }
   return result
 }
+
+export async function getContentTree(
+  path?: string,
+  ref: string = 'main'
+): Promise<GitContentDataResponse[]> {
+  try {
+    let data = (
+      await octokit.repos.getContent({
+        ...repoData,
+        path: path ?? 'gcn',
+        ref,
+      })
+    ).data as GitContentDataResponse[]
+
+    for (let item of data) {
+      if (item.type == 'dir' || item.type == 'submodule') {
+        item.children = await getContentTree(item.path, ref)
+      }
+    }
+
+    return data
+  } catch (error) {
+    return []
+  }
+}
```

Comment:
I'm troubled by the performance implications of recursively walking the entire directory tree. It occurs to me that if the schema browser side nav supports expanding _only a single directory at a time_, then we only need to return the listing of a single directory in any one loader invocation.